### PR TITLE
Migrate Feedback module from `Java` to `Kt`

### PR DESCRIPTION
**Pull request by neeldoshii**
* Fixes: #5984
* Part of: https://github.com/commons-app/apps-android-commons/issues/5928

### CODE REVIEW

**Comment by neeldoshii** [comment]
image
Tested Link : https://commons.wikimedia.org/wiki/Commons:Mobile_app/Feedback#Feedback_from_Devanonymous_for_version_5.0.2-debug-Migrate-Feedback-Module-from-%60java%60-to-%60kt%60~6a508be92_on_2024/12/01_23:44:45

File: app/src/main/java/fr/free/nrw/commons/feedback/FeedbackDialog.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import fr.free.nrw.commons.utils.DeviceInfoUtil.getDeviceManufacturer
+import fr.free.nrw.commons.utils.DeviceInfoUtil.getDeviceModel
+
+class FeedbackDialog(
+    context: Context,
+    private val onFeedbackSubmitCallback: OnFeedbackSubmitCallback) : Dialog(context) {
+    private var _binding: DialogFeedbackBinding? = null
+    private val binding get() = _binding!!
+    // TODO("Remove Deprecation") //'fromHtml(String!): Spanned!' is deprecated. Deprecated in Java
+    @Suppress("DEPRECATION")
```
Comment (neeldoshii) [inline app/src/main/java/fr/free/nrw/commons/feedback/FeedbackDialog.kt:27]: Should I create an seperate issue for this focusing on deprecation?

File: app/src/main/java/fr/free/nrw/commons/feedback/FeedbackDialog.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        _binding = DialogFeedbackBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        binding.feedbackDestination.text = feedbackDestinationHtml
+        binding.feedbackDestination.movementMethod = LinkMovementMethod.getInstance()
+        // TODO("DEPRECATION") // 'SOFT_INPUT_ADJUST_RESIZE: Int' is deprecated. Deprecated in Java
+        @Suppress("DEPRECATION")
+        window?.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE)
```
Comment (neeldoshii) [inline app/src/main/java/fr/free/nrw/commons/feedback/FeedbackDialog.kt:40]: same.

**Comment by neeldoshii** [comment]
Test Case passed and Ready for review.

File: app/src/main/java/fr/free/nrw/commons/feedback/FeedbackDialog.kt (hunk 2 above)
Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/feedback/FeedbackDialog.kt:40]: Yes thanks! 🙂

**Comment by neeldoshii** [comment]
image
https://commons.wikimedia.org/wiki/Commons:Mobile_app/Feedback#Feedback_from_Devanonymous_for_version_5.0.2-debug-Migrate-Feedback-Module-from-%60java%60-to-%60kt%60~6a508be92_on_2024/12/01_23:44:45

File: app/src/main/java/fr/free/nrw/commons/feedback/FeedbackDialog.kt (hunk 2 above)
Comment (neeldoshii) [reply app/src/main/java/fr/free/nrw/commons/feedback/FeedbackDialog.kt:42]: Created separate issue for it.

**Comment by nicolas-raoul** [comment]
In the future would you mind avoiding quotes in branch names? Some environment may be incompatible with them, for instance my command-line gradle chokes on it:
```
Execution failed for task ':app:installProdDebug'.
> java.util.concurrent.ExecutionException: com.android.builder.testing.api.DeviceException: com.android.ddmlib.InstallException: Unknown failure: /system/bin/sh: java: inaccessible or not found
  /system/bin/sh: kt: inaccessible or not found
  ```
  No worries for this time, I just renamed the branch locally. :-)

**Comment by nicolas-raoul** [comment]
Tested, works as expected. Thanks! 🙂

**Review by nicolas-raoul — APPROVED** [review]

**Pull request merged**